crates/bevy_mesh/src/mesh.rs

```diff
@@ -1362,23 +2135,102 @@ impl Mesh {
             })
         }
     }
+
+    /// Extracts the mesh vertex, index and morph target data for GPU upload.
+    /// This function is called internally in render world extraction, it is
+    /// unlikely to be useful outside of that context.
+    ///
+    /// Returns an error if the mesh data has been extracted to `RenderWorld`.
+    pub fn take_gpu_data(&mut self) -> Result<Self, MeshAccessError> {
+        let attributes = self.attributes.extract()?;
+        let indices = self.indices.extract()?;
+        #[cfg(feature = "morph")]
+        let morph_targets = self.morph_targets.extract()?;
+        #[cfg(feature = "morph")]
+        let morph_target_names = self.morph_target_names.extract()?;
+
+        // store the aabb extents as they cannot be computed after extraction
+        if let Some(MeshAttributeData {
+            values: VertexAttributeValues::Float32x3(position_values),
+            ..
+        }) = attributes.as_ref()?.get(&Self::ATTRIBUTE_POSITION.id)
+        {
```

Comment:
```suggestion
        }) = attributes.as_ref()?.get(&Self::ATTRIBUTE_POSITION.id)
            && !position_values.is_empty()     
        {
```

Not 100% sure it's actually possible to make and extract a mesh with an empty position attribute, but seems reasonable to avoid any chance of panicking.